### DISCUSSION TITLE
How to handle table selection event (row checked/unchecked)

### DISCUSSION THREAD

**Discussion by taltshaked** [ant-design-blazor/ant-design-blazor#3016] [Q&A]
Hi,
I need a way to capture the selection change event (e.g. row get checked/unchecked) so I can change the enable state of buttons (binding the Disabled property to the selectedRows array doesn't trigger the enable state so I need to call StateHasChanged upon the selection event I need to capture).

Please help,
Thanks

**Comment by ElderJames** [comment]
Thanks for contacting us, please try binding `SelectedRowsChanged` event callback parameter.

**Reply by taltshaked** [reply]
Thank you. Using `SelectedRowsChanged` instead of binding the selected rows to an array (as done previously) does work BUT only if my buttons are outside the table (i.e. not in the `TitleTemplate` tag>) so functionality wise, it works but also introduces layout issue (I want the buttons to be as close as possible on top of the table, which is why they were part of the `TitleTemplate`).
I also don't know if the fact the `Disabled` binding working only outside of the table tag means a behavior bug or not.

**Reply by ElderJames** [reply]
Could you provide a demo? I would like to try making better.

**Reply by taltshaked** [reply]
Somehow I cannot repro this so I'll let you know if that happens again.
I'll let you know if that happens again.
Thank you

**Reply by JesperOlsen** [reply]
I've had the same issue as OP.
Placing a search-textbox and actions-buttons in the title-template did not update correctly when their visibility was dependent on there being any selected rows. The render-code was not called for the title-template no matter how many times I called StateHasChanged on my outer component. So I moved those items(textbox+buttons) outside the Table with then no problems. I did not have any issues with layout because of this as OP.

I also had to forego the SelectedRows-property as it seems to be fired in a wrong sequence compared to rendering. The SelectedRows-property is also not updated when you use the "(un)select all"-checkbox. at the top of the selection-column. So I used the SelectedRowsChanged event instead.

When clicking (on row-checkbox) to change the selection of a row the OnRowClick-event is fired which it should not. Looking at the repo-code it seems a couple of onclick:stoppropagation="true" is required to stop the click from propagating from the checkbox to the RowClick-event.
